cmd/swarm/access.go

```diff
@@ -110,23 +110,33 @@ func accessNewACT(ctx *cli.Context) {
 	}
 
 	var (
-		ae          *api.AccessEntry
-		actManifest *api.Manifest
-		accessKey   []byte
-		err         error
-		ref         = args[0]
-		grantees    = []string{}
-		actFilename = ctx.String(SwarmAccessGrantKeysFlag.Name)
-		privateKey  = getPrivKey(ctx)
-		dryRun      = ctx.Bool(SwarmDryRunFlag.Name)
+		ae                      *api.AccessEntry
+		actManifest             *api.Manifest
+		accessKey               []byte
+		err                     error
+		ref                     = args[0]
+		grantees                = []string{}
+		actFilename             = ctx.String(SwarmAccessGrantKeysFlag.Name)
+		encryptPasswordFilename = ctx.String(utils.PasswordFileFlag.Name)
+		encryptPassword         = ""
+		privateKey              = getPrivKey(ctx)
+		dryRun                  = ctx.Bool(SwarmDryRunFlag.Name)
 	)
 
 	bytes, err := ioutil.ReadFile(actFilename)
 	if err != nil {
 		utils.Fatalf("had an error reading the grantee public key list")
 	}
 	grantees = strings.Split(string(bytes), "\n")
-	accessKey, ae, actManifest, err = api.DoACTNew(ctx, privateKey, salt, grantees)
+
+	if encryptPasswordFilename != "" {
+		bytes, err = ioutil.ReadFile(encryptPasswordFilename)
+		if err != nil {
+			utils.Fatalf("could not read password filename: %v", err)
+		}
+		encryptPassword = string(bytes)
+	}
+	accessKey, ae, actManifest, err = api.DoACTNew(ctx, privateKey, salt, grantees, encryptPassword)
```

Comment:
I dont get this. Is the file not supposed to contain passwords one per line analogous to grantees file?
I cannot find the logic of split and iterate like it is on 138 for grantees....

Reply:
it wasn't crystal clear that it was password**s** and not passwor**d** :)
in any case, in regard also to your previous comment - i'm implicitly assuming we should accommodate a scenario where just passwords are provided and no grantee public keys are provided